### PR TITLE
RFC: Modifiable cache experimentation.

### PR DESCRIPTION
**Pull request by angelnikolov**
For all who wanted a way to modify cache, now you have it. Let me know if you like the API.
Basically, you'd use it like:
```tsx
const cacheModifier = new Subject<any>();
  @Cacheable({
    storageStrategy: InMemoryStorageStrategy,
    cacheModifier
  })
  getMutableData(parameter: string) {
    return this.mockServiceCall(parameter);
}
```

Then, when you want to modify the cache, you'd emit through `cacheModifier` a callback: 
```typescript
cacheModifier.next((allCachedItems: any[]) => {
   const specificCachedItem = allCachedItems.find(cacheItem => cacheItem.parameters[0] === 'whatever...).response.payload = 'modified item';
    return allCachedItems;
});
```
The return value of this callback will be automatically stored for you in the cache, against the `cacheKey` of this method.
cc. @matthew2564 @dmbaker90 @Stefan-Kosker @muratcorlu @atmdev 
Ref: https://github.com/angelnikolov/ts-cacheable/issues/95, https://github.com/angelnikolov/ts-cacheable/issues/45, https://github.com/angelnikolov/ts-cacheable/issues/83, https://github.com/angelnikolov/ts-cacheable/issues/94

### CODE REVIEW

**Comment by Stefan-Kosker** [comment]
Damn dude seriously, you are a beast! I wished, many other open source library developers would take an example from you! Seriously.

I have one question: I just don't get where this cacheModifier method gets called. Or how I can access to it. Maybe I just dont understand it because it is a bit late. But would still be thankful for an answer.

**Comment by angelnikolov** [comment]
@Stefan-Kosker Thanks for the kind words!
So basically, you'd use it similarly to the `cacheBusterNotifier`.
You create a plain subject somewhere in your code (you can also export it from a common place or whatever), like:
```
const cacheModifier = new Subject();
```

Then you tell the `@Cacheable` decorator, that you want to use this `cacheModifier` subject to tell the cache **when** to  update and **how** to update.
`When` and `How` are important, because whatever callback you emit through the `Subject` will be called with your cache as the parameter and return the modified version.
Basically `cacheModifier.next(cache => {modify the cache here... return cache;});`
Let me know if you got more questions.
